packages/wdio-sauce-service/src/launcher.js

```diff
@@ -1,59 +1,75 @@
+import { promisify } from 'util'
+import { performance, PerformanceObserver } from 'perf_hooks'
+
 import logger from '@wdio/logger'
 import SauceConnectLauncher from 'sauce-connect-launcher'
 
 const log = logger('@wdio/sauce-service')
 export default class SauceLauncher {
+    constructor (options) {
+        this.options = options
+    }
+
     /**
      * modify config and launch sauce connect
      */
-    onPrepare (config, capabilities) {
-        if (!config.sauceConnect) {
+    async onPrepare (config, capabilities) {
+        if (!this.options.sauceConnect) {
             return
         }
 
+        const sauceConnectOpts = Object.assign({ noAutodetect: true }, this.options.sauceConnectOpts)
+        const sauceConnectTunnelIdentifier = (
+            sauceConnectOpts.tunnelIdentifier ||
+            /**
+             * generate random identifier if not provided
+             */
+            `SC-tunnel-${Math.random().toString().slice(2)}`)
+
         this.sauceConnectOpts = Object.assign({
             username: config.user,
             accessKey: config.key,
-            logger: log.debug
-        }, config.sauceConnectOpts)
+            logger: log.debug,
+            tunnelIdentifier: sauceConnectTunnelIdentifier
+        }, sauceConnectOpts)
```

Comment:
Instead of using `Object.assign` can we use spread operator to clone the object?